app/domain/people/duplicate_locator.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2012-2020, Jungwacht Blauring Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class People::DuplicateLocator
+  def run
+    Person.find_each do |person|
+      duplicate = person_duplicate_finder.find({ first_name: person.first_name,
```

Comment:
extract to #find_duplicate(person)

```
duplicate = find_duplicate(person)

next unless duplicate

create ... unless present?
```

on second thoughts ... in that case you could just use PersonDuplicate.find_or_create_by!(...)

sry if I pointed you in the wrong direction here :wink: